app/nasa_collaboration/page.js

```diff
@@ -1,30 +1,59 @@
 "use client";
 
-import React, { useState, useEffect } from 'react';
-import styles from './page.module.css';
+import React, { useState, useEffect } from "react";
+import styles from "./page.module.css";
 
 // Read "/app/nasa_collaboration/README.md" for more info about the API_KEY
 // You need a proper API_KEY for the requests to work
-const API_KEY = 'API_KEY';
+const API_KEY = "nDgmi28bavvf27vTGSZ8JoPwd4BfLJtrJP6DKhgC";
 
 const NASA_URLs = {
   astronomyPicOfTheDay: `https://api.nasa.gov/planetary/apod?api_key=${API_KEY}`,
   marsRoverPhoto: `https://api.nasa.gov/mars-photos/api/v1/rovers/curiosity/photos?earth_date=2015-6-3&api_key=${API_KEY}`,
 };
 
+const RoverPhoto =({src, date, roverName}) => {
+  return (
+    <div >
+      <h4>{roverName}</h4>
+      <p>Date: {date}</p>
+      <img
+        className={styles.nasaPicOfTheDayImg}
+        src={src}
+        alt={`${roverName} Rover Photo`}
+      />
+    </div>
+  );
+}
 export const NasaCollaboration = () => {
   const [dailyImg, setDailyImg] = useState({});
   const [roverPhoto, setRoverPhoto] = useState({});
 
   useEffect(() => {
+  
     const fetchRoverPhotos = async () => {
-      const roverPhotoResponse = await fetch(NASA_URLs.marsRoverPhoto).then(response => response.json());
-      setRoverPhoto(roverPhotoResponse);
+      try {
+        const roverPhotoResponse = await fetch(NASA_URLs.marsRoverPhoto).then(
+          (response) => response.json()
+        );
+        setRoverPhoto(roverPhotoResponse);
+      } catch (error) {
+        console.error("Error fetching Mars Rover photos:", error);
+      }; 
+    }
+const fetchDailyImg = async () => {
+    try {
+      const dailyImgResponse = await fetch(NASA_URLs.astronomyPicOfTheDay).then(
+        (response) => response.json()
+      );
+      setDailyImg(dailyImgResponse);
+    } catch (error) {
+      console.error("Error fetching Astronomy Picture of the Day:", error);
```

Comment:
How is this error communicated to the user? I.e., how will the user know something went wrong? :)

Reply:
I have added a use State to control how the error is being displayed to the users